index.js

```diff
@@ -0,0 +1,180 @@
+        });
+    }
+
+    /**
+     * Initialize the jenkins client with crumb
+     * @method initJenkinsClient
+     * @param  crumb                        CSRF token from jenkins api
+     * @param  {Function} callback          Callback with error and jenkins client object
+     */
+    initJenkinsClient(crumb, callback) {
+        const data = JSON.parse(crumb);
+
+        const jenkinsClient = jenkins({
+            baseUrl: `http://${this.username}:${this.password}@${this.host}:8080`,
+            headers: {
+                [data.crumbRequestField]: data.crumb
+            }
+        });
+
+        if (jenkinsClient) return callback(null, jenkinsClient);
+
+        return callback(new Error('Failed to instantiate jenkins client'));
+    }
+
+    /**
+     * Create a jenkins job and start the build
+     * @method createJob
+     * @param  {Object}   config            A configuration object
+     * @param  {String}   config.buildId    ID for the build and also name of the job in jenkins
+     * @param  {String}   config.jobId      ID for the job
+     * @param  {String}   config.pipelineId ID for the pipeline
+     * @param  {String}   config.container  Container for the build to run in
+     * @param  {String}   config.scmUrl     Scm URL to use in the build
+     * @param  {Function} callback          Callback with null if successful otherwise error
+     */
+    _start(config, callback) {
+        const fakeConfigPath = path.resolve(__dirname, './config/test-job.xml');
+        const xml = fs.readFileSync(fakeConfigPath, 'utf-8');
+
+        async.waterfall([
+            this.getCrumb.bind(this),
+            this.initJenkinsClient.bind(this),
+            (jenkinsClient, cb) => {
+                const create = jenkinsClient.job.create.bind(jenkinsClient.job);
+
+                return create(config.buildId, xml, (err) => {
+                    cb(err, jenkinsClient);
+                });
+            },
+            (jenkinsClient, cb) => {
+                const build = jenkinsClient.job.build.bind(jenkinsClient.job);
+
+                return build(config.buildId, cb);
+            }
+        ], (error) => {
+            if (error) return callback(new Error(error.message));
+
+            return callback(null);
+        });
+    }
+
+    /**
+     * Stop the build
+     * @method createJob
+     * @param  {Object}   config            A configuration object
+     * @param  {String}   config.buildId    ID for the build and also name of the job in jenkins
+     * @param  {Function} callback          Callback with null if successful otherwise error
+     */
+    _stop(config, callback) {
+        async.waterfall([
+            this.getCrumb.bind(this),
+            this.initJenkinsClient.bind(this),
+            (jenkinsClient, cb) => {
+                const get = jenkinsClient.job.get.bind(jenkinsClient.job);
+
+                return get(config.buildId, (err, data) => {
+                    try {
```

Comment:
It looks like you're using the `try - catch` because `data.lastBuild` may not exist. Are you trying to catch a "Cannot read property of undefined" error?